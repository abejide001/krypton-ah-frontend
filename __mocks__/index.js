import author from './assets/author.png';
import featuredImage from './assets/featuredImage.png';
import user1 from './assets/man2.png';
import user2 from './assets/man3.png';

const ARTICLE_TITLE = 'How to learn nodejs in 2 minutes',
  ARTICLE_AUTHOR = 'By Mark Adeniran',
  ARTICLE_DATE = 'January 02, 2019',
  ARTICLE_READ_TIME = '3 Mins read';
const ARTICLE_CONTENT = `Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget
dolor.Aenean massa strong.Cum sociis natoque penatibus et magnis dis parturient
montes, nascetur ridiculus mus.Donec quam felis, ultricies nec, pellentesque eu,
   pretium quis, sem.Nulla consequat massa quis enim.Donec pede justo, fringilla vel,
      aliquet nec, vulputate eget, arcu.In enim justo, rhoncus ut, imperdiet a, venenatis
vitae, justo.Nullam dictum felis eu pede link mollis pretium.Integer tincidunt.Cras
dapibus.Vivamus elementum semper nisi.Aenean vulputate eleifend tellus.Aenean leo
ligula, porttitor eu, consequat vitae, eleifend ac, enim.Aliquam lorem ante, dapibus
   in, viverra quis, feugiat a, tellus.Phasellus viverra nulla ut metus varius laoreet.
Quisque rutrum.Aenean imperdiet.Etiam ultricies nisi vel augue.Curabitur
ullamcorper ultricies nisi.Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
Aenean commodo ligula eget dolor.Aenean massa strong.Cum sociis natoque penatibus et
magnis dis parturient montes, nascetur ridiculus mus.Donec quam felis, ultricies nec,
   pellentesque eu, pretium quis, sem.Nulla consequat massa quis enim.Donec pede justo,
      fringilla vel, aliquet nec, vulputate eget, arcu.In enim justo, rhoncus ut, imperdiet
a, venenatis vitae, justo.Nullam dictum felis eu pede link mollis pretium.Integer
tincidunt.Cras dapibus.Vivamus elementum semper nisi.Aenean vulputate eleifend
tellus.Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim.Aliquam
lorem ante, dapibus in, viverra quis, feugiat a, tellus.Phasellus viverra nulla ut
metus varius laoreet.Quisque rutrum.Aenean imperdiet.Etiam ultricies nisi vel
augue.Curabitur ullamcorper ultricies nisi.Lorem ipsum dolor sit amet, consectetuer adipiscing elit.Aenean commodo ligula eget
dolor.Aenean massa strong.Cum sociis natoque penatibus et magnis dis parturient
montes, nascetur ridiculus mus.Donec quam felis, ultricies nec, pellentesque eu,
   pretium quis, sem.Nulla consequat massa quis enim.Donec pede justo, fringilla vel,
      aliquet nec, vulputate eget, arcu.In enim justo, rhoncus ut, imperdiet a, venenatis
vitae, justo.Nullam dictum felis eu pede link mollis pretium.Integer tincidunt.Cras
dapibus.Vivamus elementum semper nisi.Aenean vulputate eleifend tellus.Aenean leo
ligula, porttitor eu, consequat vitae, eleifend ac, enim.Aliquam lorem ante, dapibus
   in, viverra quis, feugiat a, tellus.Phasellus viverra nulla ut metus varius laoreet.
Quisque rutrum.Aenean imperdiet.Etiam ultricies nisi vel augue.Curabitur
ullamcorper ultricies nisi.Lorem ipsum dolor sit amet, consectetuer adipiscing elit.
Aenean commodo ligula eget dolor.Aenean massa strong.Cum sociis natoque penatibus et
magnis dis parturient montes, nascetur ridiculus mus.Donec quam felis, ultricies nec,
   pellentesque eu, pretium quis, sem.Nulla consequat massa quis enim.Donec pede justo,
      fringilla vel, aliquet nec, vulputate eget, arcu.In enim justo, rhoncus ut, imperdiet
a, venenatis vitae, justo.Nullam dictum felis eu pede link mollis pretium.Integer
tincidunt.Cras dapibus.Vivamus elementum semper nisi.Aenean vulputate eleifend
tellus.Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim.Aliquam
lorem ante, dapibus in, viverra quis, feugiat a, tellus.Phasellus viverra nulla ut
metus varius laoreet.Quisque rutrum.Aenean imperdiet.Etiam ultricies nisi vel
augue.Curabitur ullamcorper ultricies nisi.`;
const COMMENT_TEXT = 'This will be great for business reports. I will definitely download this.',
  NUMBER_OF_LIKES = '23',
  NUMBER_OF_DISLIKES = '12',
  COMMENT_TIME = '3 days ago',
  NUMBER_OF_THREADS = '20 threads';

const commentsArray = [
  {
    id: 1,
    commentText: COMMENT_TEXT,
    numberOfLikes: NUMBER_OF_LIKES,
    numberOfDisikes: NUMBER_OF_DISLIKES,
    commentTime: COMMENT_TIME,
    numberOfThreads: NUMBER_OF_THREADS,
    avatar: user1
  },
  {
    id: 2,
    commentText: COMMENT_TEXT,
    numberOfLikes: NUMBER_OF_LIKES,
    numberOfDisikes: NUMBER_OF_DISLIKES,
    commentTime: COMMENT_TIME,
    numberOfThreads: NUMBER_OF_THREADS,
    avatar: user2
  },
  {
    id: 3,
    commentText: COMMENT_TEXT,
    numberOfLikes: NUMBER_OF_LIKES,
    numberOfDisikes: NUMBER_OF_DISLIKES,
    commentTime: COMMENT_TIME,
    numberOfThreads: NUMBER_OF_THREADS,
    avatar: user1
  }
];

export {
  author,
  commentsArray,
  featuredImage,
  user1,
  user2,
  ARTICLE_TITLE,
  ARTICLE_AUTHOR,
  ARTICLE_DATE,
  ARTICLE_READ_TIME,
  ARTICLE_CONTENT
};
