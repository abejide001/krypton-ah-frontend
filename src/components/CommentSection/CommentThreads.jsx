/* eslint-disable react/destructuring-assignment */
import React, { Component, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Comment, Container, Divider, Image } from 'semantic-ui-react';
import moment from 'moment';


import * as actions from '../../actions/commentAction/commentActions';

import './styles/CommentSection.scss';
import dislike from '../../images/dislike.svg';
import like from '../../images/like.svg';

class CommentThreads extends Component {
  constructor(props) {
    super(props);
    this.state = {
      articleId: this.props.articleId,
      mainCommentId: this.props.mainCommentId,
      count: this.props.count
    };
  }

  componentDidMount() {
    this.props.getComments(this.state.articleId, this.state.mainCommentId);
    this.props.getCommentLikes();
  }

  getCommentActions = (numberOfLikes, numberOfDisikes, commentTime) => (
    <Comment.Actions>
      <Comment.Action>
        <Image src={like} />
        {numberOfLikes}
      </Comment.Action>
      <Comment.Action>
        <Image src={dislike} />
        {numberOfDisikes}
      </Comment.Action>
      <Comment.Metadata>
        <span>{commentTime}</span>
      </Comment.Metadata>
    </Comment.Actions>
  );

  getComment = (comment, commentLike) => (
    <Comment>
      <Comment.Avatar as="a" src={comment.avatar} />
      <Comment.Content>
        <Comment.Text>{comment.comment}</Comment.Text>
        {this.getCommentActions(
          commentLike.like,
          commentLike.dislike,
          moment.utc(comment.createdAt, 'YYYY-MM-DDTHH:mm:ss').fromNow()
        )}
      </Comment.Content>
    </Comment>
  );

  render() {
    const { threadsArray } = this.props.comment;
    const { commentLike } = this.props.comment;
    if (this.state.count > 0) {
      return (
        <Container>
          <div className="commentsContainer">
            <Comment.Group>
              {threadsArray.map(comment => (
                <Fragment key={comment.id}>
                  {this.getComment(comment, commentLike)}
                  <Divider className="commentDivider" />
                </Fragment>
              ))}
            </Comment.Group>
          </div>
        </Container>
      );
    }
    return null;
  }
}

export const mapStateToProps = state => ({
  comment: state.commentReducer
});

export const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CommentThreads);

CommentThreads.propTypes = {
  articleId: PropTypes.string.isRequired,
  mainCommentId: PropTypes.string,
  comment: PropTypes.object.isRequired,
  threadsArray: PropTypes.array,
  getComments: PropTypes.func.isRequired,
  getCommentLikes: PropTypes.func.isRequired
};

CommentThreads.defaultProps = {
  threadsArray: [],
  mainCommentId: null
};
