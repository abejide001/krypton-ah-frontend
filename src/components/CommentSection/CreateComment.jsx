/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Form } from 'semantic-ui-react';

import { createComment } from '../../helpers/axiosHelper/commentRequests';

class CreateComment extends Component {
  constructor(props) {
    super(props);
    this.state = {
      articleId: this.props.articleId,
      mainCommentId: this.props.mainCommentId,
      commentMessage: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(e) {
    e.preventDefault();

    const { articleId, mainCommentId, commentMessage } = this.state;
    createComment(articleId, commentMessage, mainCommentId);

    this.setState(state => ({
      ...state,
      commentMessage: ''
    }));
  }

  handleChange(e) {
    const { name, value } = e.target;
    this.setState(state => ({
      ...state,
      [name]: value
    }));
  }

  render() {
    return (
      <Form onSubmit={this.handleSubmit}>
        <Form.TextArea
          name="commentMessage"
          value={this.state.commentMessage}
          onChange={this.handleChange} />
        <Button content="Post" labelPosition="left" icon="edit" primary />
      </Form>
    );
  }
}

export default CreateComment;

CreateComment.propTypes = {
  articleId: PropTypes.string.isRequired,
  mainCommentId: PropTypes.string
};

CreateComment.defaultProps = {
  mainCommentId: null
};
