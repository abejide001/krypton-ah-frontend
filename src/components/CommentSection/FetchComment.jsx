/* eslint-disable react/destructuring-assignment */
import React, { Component, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Comment, Header, Container, Divider } from 'semantic-ui-react';

import * as actions from '../../actions/commentAction/commentActions';

class FetchComment extends Component {
  constructor(props) {
    super(props);
    this.state = {
      articleId: this.props.articleId,
      mainCommentId: this.props.mainCommentId
    };
  }

  componentDidMount() {
    console.log(this.state);
    this.props.getComments(this.state.articleId, this.state.mainCommentId);
  }

  render() {
    console.log(this.props.comment.commentsArray)
    const { commentsArray } = this.props.comment;
    return (
      <Container>
        <div className="commentsContainer">
          <Comment.Group>
            <Header as="h3" className="commentsHeader">
              Comments
            </Header>
            {commentsArray.map(comment => (
              <Fragment key={comment.id}>
                {comment.comment}
                <Divider className="commentDivider" />
              </Fragment>
            ))}
          </Comment.Group>
          <p className="moreComments">More Comments</p>
        </div>
      </Container>
    );
  }
}

export const mapStateToProps = state => ({
  comment: state.commentReducer
});

export const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FetchComment);

FetchComment.propTypes = {
  articleId: PropTypes.string.isRequired,
  mainCommentId: PropTypes.string.isRequired,
  comment: PropTypes.object.isRequired,
  commentsArray: PropTypes.array.isRequired,
  getComments: PropTypes.func.isRequired
};
