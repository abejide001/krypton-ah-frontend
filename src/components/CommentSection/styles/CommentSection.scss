@import '../../../styles/partials/fonts';

.ui.header {
  font-family: 'Roboto', sans-serif;
}
.ui.header .commentsHeader {
  font-size: 1.5rem;
  text-transform: uppercase;
}
.commentsContainer {
  width: 100%;
  margin: 2% 7% 10% 6%;
  font-weight: 100;
  font-size: 0.7rem;
}
.commentDivider {
  width: 80%;
}
.showThreads {
  display: inline;
}
.ui.comments {
  .comment {
    .actions {
      a {
        &:last-child {
          margin-left: 17%;
          font-size: 0.7rem;
          text-transform: capitalize;
        }
      }
    }
    .text {
      font-size: 0.9rem;
    }
  }
}
.moreComments {
  float: right;
  margin-top: -3%;
  margin-right: 20%;
  cursor: pointer;
}

.replyButton {
  font-size: 0.7rem;
  padding-left: 3%;
}
