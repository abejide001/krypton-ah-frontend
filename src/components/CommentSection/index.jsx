import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import { Comment, Header, Container, Divider, Image } from 'semantic-ui-react';
import './styles/CommentSection.scss';
import dislike from '../../images/dislike.svg';
import like from '../../images/like.svg';

export default class CommentSection extends Component {
  getComment = comment => (
    <Comment>
      <Comment.Avatar as="a" src={comment.avatar} />
      <Comment.Content>
        <Comment.Text>{comment.commentText}</Comment.Text>
        {this.getCommentActions(
          comment.numberOfLikes,
          comment.numberOfDisikes,
          comment.commentTime,
          comment.numberOfThreads
        )}
      </Comment.Content>
    </Comment>
  );

  getCommentActions = (numberOfDisikes, numberOfLikes, numberOfThreads, commentTime) => (
    <Comment.Actions>
      <Comment.Action>
        <Image src={like} />
        {numberOfLikes}
      </Comment.Action>
      <Comment.Action>
        <Image src={dislike} />
        {numberOfDisikes}
      </Comment.Action>
      <Comment.Metadata>
        <span>{commentTime}</span>
      </Comment.Metadata>
      <Comment.Action className="replyButton">Reply</Comment.Action>
      <Comment.Action className="showThreads">{numberOfThreads}</Comment.Action>
    </Comment.Actions>
  );

  render() {
    const { commentsArray } = this.props;
    return (
      <Container>
        <div className="commentsContainer">
          <Comment.Group>
            <Header as="h3" className="commentsHeader">
              Comments
            </Header>
            {commentsArray.map(comment => (
              <Fragment key={comment.id}>
                {this.getComment(comment)}
                <Divider className="commentDivider" />
              </Fragment>
            ))}
          </Comment.Group>
          <p className="moreComments">More Comments</p>
        </div>
      </Container>
    );
  }
}

CommentSection.propTypes = { commentsArray: PropTypes.array };
CommentSection.defaultProps = { commentsArray: [] };
