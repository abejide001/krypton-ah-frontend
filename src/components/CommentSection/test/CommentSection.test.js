import React from 'react';
import { shallow } from 'enzyme';
import CommentSection from '../index';
import { commentsArray } from '../../../../__mocks__';

describe('CommentSection', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<CommentSection commentsArray={commentsArray} />);
    expect(wrapper).toMatchSnapshot();
  });
});
