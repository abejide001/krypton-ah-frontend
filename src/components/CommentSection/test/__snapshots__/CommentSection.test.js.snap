// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentSection should match snapshot 1`] = `
<Container>
  <div
    className="commentsContainer"
  >
    <CommentGroup>
      <Header
        as="h3"
        className="commentsHeader"
      >
        Comments
      </Header>
      <Comment>
        <CommentAvatar
          as="a"
          src="test-file-stub"
        />
        <CommentContent>
          <CommentText>
            This will be great for business reports. I will definitely download this.
          </CommentText>
          <CommentActions>
            <CommentAction
              as="a"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              />
              12
            </CommentAction>
            <CommentAction
              as="a"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              />
              23
            </CommentAction>
            <CommentMetadata>
              <span>
                20 threads
              </span>
            </CommentMetadata>
            <CommentAction
              as="a"
              className="replyButton"
            >
              Reply
            </CommentAction>
            <CommentAction
              as="a"
              className="showThreads"
            >
              3 days ago
            </CommentAction>
          </CommentActions>
        </CommentContent>
      </Comment>
      <Divider
        className="commentDivider"
      />
      <Comment>
        <CommentAvatar
          as="a"
          src="test-file-stub"
        />
        <CommentContent>
          <CommentText>
            This will be great for business reports. I will definitely download this.
          </CommentText>
          <CommentActions>
            <CommentAction
              as="a"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              />
              12
            </CommentAction>
            <CommentAction
              as="a"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              />
              23
            </CommentAction>
            <CommentMetadata>
              <span>
                20 threads
              </span>
            </CommentMetadata>
            <CommentAction
              as="a"
              className="replyButton"
            >
              Reply
            </CommentAction>
            <CommentAction
              as="a"
              className="showThreads"
            >
              3 days ago
            </CommentAction>
          </CommentActions>
        </CommentContent>
      </Comment>
      <Divider
        className="commentDivider"
      />
      <Comment>
        <CommentAvatar
          as="a"
          src="test-file-stub"
        />
        <CommentContent>
          <CommentText>
            This will be great for business reports. I will definitely download this.
          </CommentText>
          <CommentActions>
            <CommentAction
              as="a"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              />
              12
            </CommentAction>
            <CommentAction
              as="a"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              />
              23
            </CommentAction>
            <CommentMetadata>
              <span>
                20 threads
              </span>
            </CommentMetadata>
            <CommentAction
              as="a"
              className="replyButton"
            >
              Reply
            </CommentAction>
            <CommentAction
              as="a"
              className="showThreads"
            >
              3 days ago
            </CommentAction>
          </CommentActions>
        </CommentContent>
      </Comment>
      <Divider
        className="commentDivider"
      />
    </CommentGroup>
    <p
      className="moreComments"
    >
      More Comments
    </p>
  </div>
</Container>
`;
